HKMP/Game/Client/PauseManager.cs

```diff
@@ -15,9 +16,15 @@ internal class PauseManager {
     /// The net client instance.
     /// </summary>
     private readonly NetClient _netClient;
+    
+    /// <summary>
+    /// Hook for time scale changes.
+    /// </summary>
+    private readonly IClientManager.SetTimeScale _onSetTimeScale;
 
-    public PauseManager(NetClient netClient) {
+    public PauseManager(NetClient netClient, IClientManager.SetTimeScale onSetTimeScale) {
```

Comment:
I think it makes more sense to expose the event for the time scale changing in this class instead. Now we are just juggling a delegate around just to be able to call the event in the `ClientManager` class.